skt/__init__.py

```diff
@@ -518,24 +539,73 @@ def mktgz(self, clean=True):
 
         logging.info("building kernel: %s", args)
 
-        mk = subprocess.Popen(args,
-                              stdout=subprocess.PIPE,
-                              stderr=subprocess.STDOUT)
-        (stdout, stderr) = mk.communicate()
-        for line in stdout.split("\n"):
-            m = re.match("^Tarball successfully created in (.*)$", line)
-            if m:
-                tgzpath = m.group(1)
-                break
+        with io.open(self.buildlog, 'wb') as writer, \
+                io.open(self.buildlog, 'rb') as reader:
+            make = subprocess.Popen(args,
+                                    stdout=writer,
+                                    stderr=subprocess.STDOUT)
+            make_timedout = []
+
+            def stop_process(proc):
+                """
+                Terminate the process with SIGTERM and flag it as timed out.
+                """
+                if proc.poll() is None:
+                    proc.terminate()
+                    make_timedout.append(True)
+            timer = Timer(timeout, stop_process, [make])
+            timer.setDaemon(True)
+            timer.start()
+            while make.poll() is None:
+                self.append_and_log2stdout(reader.readlines(), stdout_list)
+                time.sleep(1)
+            self.append_and_log2stdout(reader.readlines(), stdout_list)
+            timer.cancel()
```

Comment:
Could you wrap the code executing while the timer is running, above, into a try-finally, and put `timer.cancel` into the `finally`, so we're sure the timer gets killed?

Reply:
Great idea! :+1: